src/mtype.d

```diff
@@ -7868,14 +7868,38 @@ public:
                 // goto L1;
             }
         }
-        s = sym.search(e.loc, ident);
```

Comment:
Yes, I think we should get rid of the default flags for `Dsymbol.search`. They are even different for Dsymbol and ScopeDSymbol :o.